crates/net/network/src/manager.rs

```diff
@@ -590,6 +590,9 @@ impl NetworkManager {
             NetworkHandleMessage::DisconnectPeer(peer_id, reason) => {
                 self.swarm.sessions_mut().disconnect(peer_id, reason);
             }
+            NetworkHandleMessage::ReConnect(socket_addr, peer_id) => {
+                self.swarm.sessions_mut().dial_outbound(socket_addr, peer_id);
```

Comment:
this isn't quite right, because the peer can not exist in the peers manager.

we need to go trough the peersmanager and add this logic there, like `add_and_connect`, similar to

https://github.com/paradigmxyz/reth/blob/f2bb8e74d9700d8a46487896e085aaecb07a720c/crates/net/network/src/peers.rs#L645-L645

but also issue a connect message to that peer if it's unconnected, like

https://github.com/paradigmxyz/reth/blob/f2bb8e74d9700d8a46487896e085aaecb07a720c/crates/net/network/src/peers.rs#L806-L806

we also need some tests for this, see peersmanager and we also want an e2e test for this as weel see tests/it/connect for ref